### PR TITLE
CMR-6899 - EDL Token support

### PR DESCRIPTION
**Pull request by jceaser**
CMR-6899 : Adding documentation about using EDL tokens and allowed token file to have comments, added jupyter notebook test

The code was already working with the new EDL tokens but no documentation existing. Also added the ability to comment out tokens in the token file making it easy to have SIT, UAT, and OPS tokens all at once.

To try out this code and too see more information on it, try the token notebook.

### CODE REVIEW

File: CMR/python/cmr/auth/token.py [hunk 1]
```diff
@@ -115,7 +121,10 @@ def token_manager(config: dict = None):
 
 def token(token_lambdas = None, config: dict = None):
     """
-    Recursively calls lambdas till a token is found
+    Recursively calls lambdas till a token is found. These lamdba functions
```
Comment (eereiter) [inline CMR/python/cmr/auth/token.py:124]: Change the word Recursively to iterate through the list until a token is found. or something like that.

File: CMR/python/demos/notebooks/tokens.ipynb [hunk 2]
```diff
@@ -14,7 +14,7 @@
    "cell_type": "markdown",
    "metadata": {},
    "source": [
-    "## Demo Safe\n",
+    "## Practice Demo Safe\n",
     "Just for this demo, I'm going to create a function that hids some of an EDL token, I don't want anyone to actually see my tokens"
```
Comment (eereiter) [inline CMR/python/demos/notebooks/tokens.ipynb:18]: misspelled hids  - should be hides.

File: CMR/python/demos/notebooks/tokens.ipynb [hunk 3]
```diff
@@ -42,7 +43,7 @@
    "metadata": {},
    "source": [
     "## Loading the library\n",
-    "This will not be needed once we have the library working through PIP, but today I'm going to use my local checkout."
+    "This will not be needed once we have the library working through PIP, but with this a local checkout can be used. In this case, the command `git clone https://github.com/nasa/eo-metadata-tools` was called in `~/src/project`."
```
Comment (eereiter) [inline CMR/python/demos/notebooks/tokens.ipynb:46]: with this, (add comma)

**Review by eereiter — CHANGES_REQUESTED** [review]

**Review by eereiter — APPROVED** [review]

**Review by mmorahan — APPROVED** [review]
Looks good to me.

**Pull request merged**